Sources/Profile-Modify.php

```diff
@@ -458,11 +458,26 @@ function $preload:		A function that is used to load data required for this eleme
 			'callback_func' => 'smiley_pick',
 			'enabled' => !empty($modSettings['smiley_sets_enable']),
 			'permission' => 'profile_extra',
-			'preload' => function() use ($modSettings, &$context, $txt, $cur_profile, $smcFunc)
+			'preload' => function() use ($modSettings, &$context, $txt, $cur_profile, $smcFunc, $settings)
 			{
 				$context['member']['smiley_set']['id'] = empty($cur_profile['smiley_set']) ? '' : $cur_profile['smiley_set'];
 				$context['smiley_sets'] = explode(',', 'none,,' . $modSettings['smiley_sets_known']);
 				$set_names = explode("\n", $txt['smileys_none'] . "\n" . $txt['smileys_forum_board_default'] . "\n" . $modSettings['smiley_sets_names']);
+
+				$filenames = '';
+				$result = $smcFunc['db_query']('', '
+					SELECT f.filename
+					FROM {db_prefix}smiley_files AS f
+						JOIN {db_prefix}smileys AS s ON (s.id_smiley = f.id_smiley)
+					WHERE s.code = {string:smiley}',
+					array(
+						'smiley' => ':)',
+					)
+				);
+				if ($smcFunc['db_num_rows']($result) !== 0)
+					list ($filenames) = $smcFunc['db_fetch_row']($result);
+				$smcFunc['db_free_result']($result);
```

Comment:
I don't get fully the idea here,
because you "search" for ``:)`` and return the filename --> okay
but the different smiley_set could got different filename for this.
And you choose "randomly" the first and drop all other filenames.

Reply:
When this is technical correct,
than a limit 1 would be nice.

Reply:
That was just a plain old mistake on my part. Good catch.